src/verify_sig.rs

```diff
@@ -0,0 +1,58 @@
+use rsa::{RsaPrivateKey, RsaPublicKey};
+use rsa::pkcs1::{DecodeRsaPrivateKey, DecodeRsaPublicKey};
+use rsa::pkcs1v15::{SigningKey, VerifyingKey};
+use rsa::signature::{SignatureEncoding, Signer, Verifier};
+use rsa::sha2::Sha256;
+use std::{fs, str};
+
+// Takes a data buffer, a private key, and a public key, to sign the data
+// with the private key and verify the data with the public key.
+// Assume that we rely on RSA PKCS1v1.5, sha256 for the hash.
+pub fn verify_sig_rsa_pkcs1(databuf: &[u8], private_key: RsaPrivateKey, public_key: RsaPublicKey) {
+    // Equivalent of:
+    //   openssl rsautl -verify -pubin -key |public_key_path}
+    //   - in |sig_data| -out |out_hash_data|
+
+    let signing_key = SigningKey::<Sha256>::new(private_key);
```

Comment:
I would suggest using prefixed names for signing/verifying key, to make it clear what algorithm is being used:
```rust
let signing_key = pkcs1v15::SigningKey::<Sha256>::new(private_key);
```

Reply:
Done, thanks.